flake.nix

```diff
@@ -0,0 +1,122 @@
+{
+  description = "PHP development environments";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    nix-shell.url = "github:loophp/nix-shell";
+    systems.url = "github:nix-systems/default";
+    # This should stay until Box 4.6.0 is not on `nixos-unstable` branch
+    nixpkgs-master.url = "github:NixOS/nixpkgs/master";
+  };
+
+  outputs = inputs@{ self, flake-parts, systems, ... }: flake-parts.lib.mkFlake { inherit inputs; } {
+    systems = import systems;
+
+    perSystem = { config, self', inputs', pkgs, system, lib, ... }:
+      let
+        php = pkgs.api.buildPhpFromComposer {
+          src = ./.;
+          php = pkgs.php82;
+        };
+      in
+      {
+        _module.args.pkgs = import self.inputs.nixpkgs {
+          inherit system;
+          overlays = [
+            inputs.nix-shell.overlays.default
+            (final: prev: {
+              box460 = import inputs.nixpkgs-master {
+                inherit system;
+              };
+            })
+          ];
+          config.allowUnfree = true;
+        };
+
+        apps = {
+          default = {
+            type = "app";
+            program = lib.getExe self'.packages.backwardcompatibilitycheck;
+          };
+
+          build-phar = {
+            type = "app";
+            program = lib.getExe self'.packages.build-phar-script;
+          };
+        };
+
+        checks = {
+          inherit (self'.packages) phar;
+        };
+
+        devShells.default = pkgs.mkShellNoCC {
+          name = "php-devshell";
+          buildInputs = [
+            php
+            php.packages.composer
+            pkgs.box460.php82.packages.box
+            self'.packages.build-phar-script
+          ];
+        };
+
+        packages = {
+          backwardcompatibilitycheck = php.buildComposerProject {
+            pname = "backwardcompatibilitycheck";
+            version = "8.x.x-dev";
+
+            src = ./.;
+
+            # This only changes when `composer.lock` is updated
+            vendorHash = "sha256-9VGaoPpJg06/n9fmSrNInQHitWxXStG74PxaJvulMwc=";
```

Comment:
This is still a blocker for merging this PR: `composer.lock` changes **daily**, so this added maintenance burden here blocks merge

Reply:
I know... there's no solution to this, sadly. And I haven't anything to propose to have it automatically updated.
There's this PR: https://github.com/Mic92/nix-update/pull/211 that once merged could be a potential solution, but as of today, nothing yet.

Reply:
Yeah, just wanted to spell out the problem here 👍